code-challenges.test.js

```diff
@@ -37,24 +67,97 @@ var fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
 var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
 // Expected output: [-823, 7, 23]
 
+// Input is an array
+// Create a function that takes in an array
+// Look at each data type in the array and determine if the typeof data is a number
+  // Return an array of only numbers
+// Look at each number in the array of only numbers and determine if the number is odd
+  // Return an new array of only odd numbers
+  // Return the new array and use .sort to arrange its numbers from least to greatest
+
+  describe ("oddSorted", () => {
+    test("given a mixed array return only the odd numbers sorted least to greatest", () => {
+      expect(oddSorted[fullArr1]).toEqual([-9, 7, 9, 199])
+      expect(oddSorted[fullArr2]).toEqual([-823, 7, 23])
+    })
+  })
+
+  // Ran test.
+  // Test failed.
+  // Error message: oddSorted is not defined
 
 // b) Create the function that makes the test pass.
 
+const oddSorted = (array) => {
+  let numbersOnly = (value) => {
+    if (typeof(value) === 'number') {
+      return value
+    }
+  }
+  let oddNumbers = array.filter((numbersOnly) %2 !==0)
+    return array.sort(oddNumbers)
+  }
+
+
+// Test failed again.
+// Error message = Received: undefined
```

Comment:
filter() is a higher order function and so will return an "undefined" without syntax to guide it. I can see you are attempting to nest functions and use those as arguments for your HoF's. It's a cool move and I think with a bit more research can totally be done. But in the mean time getting the filter syntax down will be really helpful.